File: restdocWithSwagger/src/main/java/com/example/restdocwithswagger/web/ResponseEntityException.java
package com.example.restdocwithswagger.web;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.server.ServerErrorException;

import java.rmi.ServerException;

@ControllerAdvice
public class ResponseEntityException {

    @ExceptionHandler(ServerErrorException.class)
    public ResponseEntity<BaseResponse> serverException(ServerErrorException e){
        return ResponseEntity.internalServerError().body(new BaseResponse(e.getMessage()));
    }
}

File: kotilWebSecurity/src/main/kotlin/kr/pe/deity/kotilwebsecurity/web/ResponseEntityException.kt
package kr.pe.deity.kotilwebsecurity.web

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.server.ServerErrorException

@ControllerAdvice
class ResponseEntityException {

    @ExceptionHandler(value =[ServerErrorException::class])
    fun serverException(e:ServerErrorException): ResponseEntity<BaseResponse>{

        val errorResponse = ErrorResponse(e.message);

        return ResponseEntity.internalServerError().body<BaseResponse>(errorResponse)
    }
}